src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -513,6 +514,17 @@ private static bool NeedsSeparator(SyntaxToken token, SyntaxToken next)
                 }
             }
 
+            if (token.IsKind(SyntaxKind.AsteriskToken) &&
+                next.IsKind(SyntaxKind.AsteriskToken) || token.Parent is PointerTypeSyntax)
+            {
+                return false;
+            }
+
+            if (next.IsKind(SyntaxKind.AsteriskToken) && next.Parent.HasAncestor<PointerTypeSyntax>())
```

Comment:
i think it's sufficient to just check immediate parent.

Reply:
@CyrusNajmabadi We can have multiple asterisks, in that case it's necessary to check ancestors.